app/models/concerns/hostext/search.rb

```diff
@@ -119,6 +121,17 @@ def search_by_user(key, operator, value)
         {:conditions => " hosts.id #{opts} " }
       end
 
+      def search_by_usergroup(key, operator, value)
+        clean_key = key.sub(/^.*\./,'')
+        key_name = Usergroup.connection.quote_column_name(clean_key)
+
+        condition = sanitize_sql_for_conditions(["#{key_name} #{operator} ?", value_to_sql(operator, value)])
+        groups = Usergroup.where(condition)
+        hosts = groups.map(&:hosts).flatten
+        opts  = hosts.empty? ? "< 0" : "IN (#{hosts.map(&:id).join(',')})"
+        {:conditions => " hosts.id #{opts} " }
```

Comment:
Are you sure an ext_method is needed at all? 
Even if so, why not join the user table and put the condition on it (with an added condition of `owner_type=Usergroup`) instead of loading possibly tens of thousands of hosts to memory just to pull out their id's and search on that again?